### PR TITLE
EVG-13579 Show Execution tasks tab on Task page

### PR DESCRIPTION
**Pull request by khelif96**
image

Adds the Execution tasks tab on the task page for display tasks
This PR also refactors the TasksTable Component and makes it more generic so it can be used in both the Patch/tasks tab and the Execution tasks tab in a task.
Also reduced renders and unintentional analytics requests
Also moved some files to follow the fractal pattern

### CODE REVIEW

File: src/pages/task/TaskTabs.tsx [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    defaultTab = activeTabs.indexOf(urlTab);
+  } else if (isDisplayTask) {
+    defaultTab = activeTabs.indexOf(TaskTab.ExecutionTasks);
+  } else if (totalTestCount > 0) {
+    defaultTab = activeTabs.indexOf(TaskTab.Tests);
+  }
+  const [selectedTab, setSelectedTab] = useState(defaultTab);
+
+  // This is used to keep track of the first tab transition so we dont accidently trigger an analytics event for it
+  const firstRender = useRef(true);
```
Comment (SupaJoon) [inline src/pages/task/TaskTabs.tsx:179]: another option is to track the previous `id` with `usePrevious` and only track analytics is prev is undefined and curr isn't

File: src/pages/task/TaskTabs.tsx [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+  useEffect(() => {
+    if (id) {
+      const query = new URLSearchParams(location.search);
+      history.replace(
+        `${paths.task}/${id}/${activeTabs[selectedTab]}?${query.toString()}`
```
Comment (SupaJoon) [inline src/pages/task/TaskTabs.tsx:184]: we should probably modify the `getTaskRoute` function to support tabs

File: src/components/Table/TasksTable.tsx [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+    className: "cy-task-table-col-STATUS",
+    render: renderStatusBadge,
+  },
+  {
+    title: "Base Status",
+    dataIndex: ["baseStatus"],
+    key: TaskSortCategory.BaseStatus,
+    sorter: (a, b) => a.baseStatus?.localeCompare(b.baseStatus),
+
```
Comment (SupaJoon) [inline src/components/Table/TasksTable.tsx:77]: nit: space

File: src/components/Table/TasksTable.tsx [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+    childrenColumnName="executionTasksFull"
+    expandable={{
+      onExpand: (expanded) => {
+        onExpand(expanded);
+      },
+    }}
+  />
+);
+
+const getColumnDefs = (onClickTaskLink): ColumnProps<TaskResult>[] => [
```
Comment (SupaJoon) [inline src/components/Table/TasksTable.tsx:51]: there some repeated key value pairs between the two getColumn helper functions. can we combine `title, key, dataIndex, width` and expand them within the two functions

File: src/components/Table/TasksTable.tsx [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+  onExpand,
+  onClickTaskLink,
+  sorts,
+}) => (
+  <Table
+    data-test-id="tasks-table"
+    rowKey={rowKey}
+    pagination={false}
+    columns={
+      sorts
```
Comment (SupaJoon) [inline src/components/Table/TasksTable.tsx:36]: why not treat sorts as an empty array if it's falsy?

**Review by SupaJoon — APPROVED** [review]

**Comment by khelif96** [comment]
Unrelated commit-queue tests are failing, Due to missing data.

**Pull request merged**